src/preset.ts

```diff
@@ -1530,29 +1534,33 @@ export async function expandBuildPreset(folder: string, name: string, workspaceF
     return expandedPreset;
 }
 
-async function expandBuildPresetImpl(folder: string, name: string, workspaceFolder: string, sourceDir: string, parallelJobs?: number, preferredGeneratorName?: string, allowUserPreset: boolean = false, configurePreset?: string, enableTryApplyDevEnv: boolean = true, errorHandler?: ExpansionErrorHandler): Promise<BuildPreset | null> {
-    let preset = getPresetByName(buildPresets(folder), name);
-    if (preset) {
-        return expandBuildPresetHelper(folder, preset, workspaceFolder, sourceDir, parallelJobs, preferredGeneratorName, false, enableTryApplyDevEnv, errorHandler);
-    }
-
-    if (allowUserPreset) {
-        preset = getPresetByName(userBuildPresets(folder), name);
+async function expandBuildPresetImpl(folder: string, name: string, workspaceFolder: string, sourceDir: string, parallelJobs?: number, preferredGeneratorName?: string, allowUserPreset: boolean = false, configurePreset?: string, enableTryApplyDevEnv: boolean = true, errorHandler?: ExpansionErrorHandler, inheritedByPreset?: BuildPreset): Promise<BuildPreset | null> {
+    const presetList = inheritedByPreset ? inheritedByPreset.__file!.buildPresets : buildPresets(folder);
+    const validInherit = presetList !== undefined && presetList.filter(p => p.name === name).length > 0;
+    if (validInherit) {
+        let preset = getPresetByName(buildPresets(folder), name);
         if (preset) {
-            return expandBuildPresetHelper(folder, preset, workspaceFolder, sourceDir, parallelJobs, preferredGeneratorName, true, enableTryApplyDevEnv, errorHandler);
+            return expandBuildPresetHelper(folder, preset, workspaceFolder, sourceDir, parallelJobs, preferredGeneratorName, false, enableTryApplyDevEnv, errorHandler);
         }
-    }
 
-    if (name === defaultBuildPreset.name) {
-        // Construct the default build preset every time since it should NOT be cached
-        preset = {
-            name: defaultBuildPreset.name,
-            displayName: defaultBuildPreset.displayName,
-            description: defaultBuildPreset.description,
-            jobs: parallelJobs || defaultNumJobs(),
-            configurePreset
-        };
-        return expandBuildPresetHelper(folder, preset, workspaceFolder, sourceDir, parallelJobs, preferredGeneratorName, true, enableTryApplyDevEnv, errorHandler);
+        if (allowUserPreset) {
+            preset = getPresetByName(userBuildPresets(folder), name);
+            if (preset) {
+                return expandBuildPresetHelper(folder, preset, workspaceFolder, sourceDir, parallelJobs, preferredGeneratorName, true, enableTryApplyDevEnv, errorHandler);
+            }
+        }
+
+        if (name === defaultBuildPreset.name) {
```

Comment:
This is where we're breaking the tests right now, we should also check for this case along with `validInherit` because otherwise we don't get to this if statement.